internal/consumer/partitionConsumer.go

```diff
@@ -228,3 +345,47 @@ func (p *partitionConsumer) sleep(d time.Duration) bool {
 		return true
 	}
 }
+
+func (p *rangePartitionConsumer) Start() error {
+	return p.lifecycle.Start(func() error {
+		go p.offsetLoop()
+		go p.commitLoop()
+		p.tally.Counter(metrics.KafkaPartitionStarted).Inc(1)
+		p.logger.Info("range partition consumer started", zap.Object("topicPartition", p.topicPartition))
+		return nil
+	})
+}
+
+func (p *rangePartitionConsumer) offsetLoop() {
+	p.logger.Info("range partition consumer offset loop started", zap.Object("topicPartition", p.topicPartition))
+	for {
+		select {
+		case ntf, ok := <-p.offsetRangeC:
+			p.logger.Debug("range partition consumer received offset range", zap.Object("topicPartition", p.topicPartition), zap.Object("offsetRange", ntf))
+			if !ok {
+				p.logger.Info("range partition consumer offset range channel closed")
+				p.Drain(p.options.MaxProcessingTime)
+				return
+			}
+			p.logger.Debug("range partition consumer ack mgr reset started", zap.Object("offsetRange", ntf))
+			<-p.ackMgr.Reset()
```

Comment:
Document the semantics of the rangePartitionConsumer. Specifically, call out that when a new offset range is specified, it will only be applied after all the previously consumed messages are either acked or nacked

Reply:
Q: How will this work with retryQs i.e. while you are waiting here for the ackMgr to drain, the app can nack one of the consumed messages and send it to retryQ or dlq. In that case, retryQ will still continue to serve messages from the old offset range. This may be ok, but document this behavior as well

Reply:
Added comment:
```
	// rangePartitionConsumer consumes only a specific offset range.
	// Only a single offset range can be consumed at a single time.
	// If an offset range is currently being consumed, calls to ResetOffset
	// will block until all messages in the previous offset range has been received
	// and acked/nacked.
	rangePartitionConsumer struct {
        ...
        }
````